### ISSUE TITLE
Double tap to upvote

### ISSUE THREAD

**Issue by CTalvio** [thunder-app/thunder#430]
Good idea spotted in play store reviews:

Double tap to upvote, triple tap to downvote.

Triple may be tricky to implement, as gesturedetect is quite aggressive in consuming the doubltap gesture.

Custom gesturedetection may be necessary.

The doubletap should be easy though.

**Comment by hjiangsu** [comment]
For this, where are we expecting to double/tap/triple tap?

**Comment by micahmo** [comment]
Just remember that as soon as we introduce a multi-tap anywhere, the single-tap action becomes slower (because it can't act until we've passed the double-tap threshold).

**Comment by CTalvio** [comment]
> For this, where are we expecting to double/tap/triple tap?

Anywhere on a post/comment. Neither currently has anything assigned to multiple taps.

> Just remember that as soon as we introduce a multi-tap anywhere, the single-tap action becomes slower (because it can't act until we've passed the double-tap threshold).

An acceptable trade-off for many, the user in question expressed distaste for the gestures, yet would like the minimalist appearance of no buttons.

I'm sure won't like the delay, but especially if we make this a custom gesture, we could tighten up the double/triple time gap a gap to make it less annoying. And it could still be a setting, as well.

**Comment by hjiangsu** [comment]
> Anywhere on a post/comment. Neither currently has anything assigned to multiple taps.

What about having a tap gesture on the upvote/downvote part of the post/comment? This would allow for quick vote actions while also not causing the delay of having to detect multiple taps on the comment/post

Attaching image for reference:
<img width="110" alt="image" src="https://github.com/thunder-app/thunder/assets/30667958/7c4f692f-5650-41f9-b839-ef8dc6089ec2">

**Comment by shortwavesurfer2009** [comment]
Doing multi tap gestures would also royally bone screen reader access as they rely on double tap and triple tap gestures to use them

**Comment by CTalvio** [comment]
> Doing multi tap gestures would also royally bone screen reader access as they rely on double tap and triple tap gestures to use them

Depends on how they are implemented, assuming they consume the inputs, they would take priority over the in-app gestures. This may or may not be the case, I do not know.
That said, I'd want to be able to disable this even without accessibility reasons. Perhaps even have it off by default.

**Comment by hjiangsu** [comment]
Yeah, I think that any multi-tap gesture should be accompanied by another alternative that does not rely on them (for example buttons)

I also think its a good idea to disable them by default, and allow options to enable/disable them
